Tests/ML/test_config_helpers.py

```diff
@@ -120,3 +123,21 @@ def test_equally_weighted_classes_fails(num_fg_clases: int, background_weight: O
     classes = [""] * num_fg_clases
     with pytest.raises(ValueError):
         equally_weighted_classes(classes, background_weight)
+
+
+def test_custom_dataset_csv(test_output_dirs: OutputFolderForTests) -> None:
+    test_csv = "test_dataset.csv"
+    dataset_csv_path = test_output_dirs.root_dir / test_csv
+    dataset_csv_path.write_text("""subject,channel,filePath""")
+    model_config = SegmentationModelBase(should_validate=False)
+    model_config.local_dataset = test_output_dirs.root_dir
+    model_config.dataset_csv = test_csv
+    dataframe = model_config.read_dataset_if_needed()
+    assert dataframe is not None
+
+    ml_util.validate_dataset_paths(model_config.local_dataset,
+                                   model_config.dataset_csv)
+    os.remove(dataset_csv_path)
+    with pytest.raises(ValueError):
+        ml_util.validate_dataset_paths(model_config.local_dataset,
+                                       model_config.dataset_csv)
```

Comment:
I find it helpful to also check if the error is really the one that we expect. When the code changes, it could fail for a completely different reason, but still raise a `ValueError`. What we do at other place is something along the lines of
```
with pytest.raises(ValueError) as ex:
    do_something()
assert "this is the expected message" in str(ex)
```

Reply:
Thanks, good suggestion!  Check of exception message added in latest commit.